sources/packages/backend/apps/api/src/services/application-restriction-bypass/application-restriction-bypass.service.ts

```diff
@@ -43,10 +70,387 @@ export class ApplicationRestrictionBypassService extends RecordDataModelService<
+      },
+      relations: {
+        studentRestriction: { restriction: true },
+        creationNote: true,
+        removalNote: true,
+        bypassCreatedBy: true,
+        bypassRemovedBy: true,
+      },
+      where: {
+        id,
+      },
+    });
+  }
+
+  /**
+   * Gets all available student restrictions to bypass for a given application.
+   * @param applicationId application id.
+   * @returns list of application restriction bypass options.
+   */
+  async getAvailableStudentRestrictionsToBypass(
+    applicationId: number,
+  ): Promise<AvailableStudentRestrictionData[]> {
+    const application = await this.applicationRepo.findOne({
+      select: {
+        id: true,
+        student: {
+          id: true,
+          studentRestrictions: {
+            id: true,
+            restriction: {
+              id: true,
+              restrictionCode: true,
+              actionType: true,
+            },
+            applicationRestrictionBypasses: {
+              id: true,
+              studentRestriction: { id: true },
+              isActive: true,
+            },
+            createdAt: true,
+          },
+        },
+        currentAssessment: {
+          id: true,
+          offering: {
+            id: true,
+            offeringIntensity: true,
+          },
+        },
+      },
+      relations: {
+        student: {
+          studentRestrictions: {
+            restriction: true,
+            applicationRestrictionBypasses: {
+              studentRestriction: true,
+            },
+          },
+        },
+        currentAssessment: { offering: true },
+      },
+      where: {
+        id: applicationId,
+        student: {
+          studentRestrictions: {
+            isActive: true,
+            restriction: {
+              actionType: ArrayOverlap([
+                RestrictionActionType.StopFullTimeBCFunding,
+                RestrictionActionType.StopFullTimeDisbursement,
+                RestrictionActionType.StopPartTimeDisbursement,
+              ]),
+            },
+          },
+        },
+      },
+    });
+
+    const allowedRestrictionActions =
+      application.currentAssessment.offering.offeringIntensity ===
+      OfferingIntensity.fullTime
+        ? [
+            RestrictionActionType.StopFullTimeBCFunding,
+            RestrictionActionType.StopFullTimeDisbursement,
+          ]
+        : [RestrictionActionType.StopPartTimeDisbursement];
+
+    const bypassedStudentRestrictionIds =
+      application.student.studentRestrictions
+        .flatMap(
+          (studentRestriction) =>
+            studentRestriction.applicationRestrictionBypasses,
+        )
+        .filter((bypass) => bypass.isActive)
+        .map((bypass) => bypass.studentRestriction.id);
+
+    const studentRestrictionsThatAreNotBypassed =
+      application.student.studentRestrictions.filter(
+        (studentRestriction) =>
+          !bypassedStudentRestrictionIds.includes(studentRestriction.id),
+      );
+
```

Comment:
This comment is for my understanding. `studentRestrictionsThatAreNotBypassed` represent an array of student restrictions for which the associated application restriction bypass records are not active. Is it correct?

Reply:
Yes, the application restriction bypass records may not be active or even not exist. If they exist and are active, we're going to remove from the available student restrictions to bypass list.